daemon/emer-network-send-provider.c

```diff
@@ -191,15 +191,19 @@ reset_network_send_data (EmerNetworkSendProvider *self)
                   error->message);
       g_error_free (error);
     }
+
+  priv->send_number = 0;
+  priv->data_cached = TRUE;
 }
 
-static gboolean
+static void
 read_network_send_data (EmerNetworkSendProvider *self)
 {
   EmerNetworkSendProviderPrivate *priv =
     emer_network_send_provider_get_instance_private (self);
+
   if (priv->data_cached)
```

Comment:
GObject definitely initializes the private struct to zeros intentionally. It may not be documented, though. Feel free to open a GNOME bug saying it should be documented ;-)